### PR TITLE
Extend URLResponse tests

### PR DESCRIPTION
**Pull request by matis-schotte**
This is my first rather small contribution to the swift project.
I tried to start with updating/adding a few tests for URLResponse.
Since I am still on High Sierra I was working on the 4.2 branch.

Please let me know if there is anything wrong or things I can do better next time.

### CODE REVIEW

**Comment by ikesyo** [comment]
Let's kicking CI.

@swift-ci Please test.

File: TestFoundation/TestURLResponse.swift [hunk 1]
```diff
@@ -31,36 +36,46 @@ class TestURLResponse : XCTestCase {
...
+    func test_MIMEType_notAvailable() {
+        let mimetype: String? = nil
+        let res = URLResponse(url: testURL, mimeType: mimetype, expectedContentLength: 0, textEncodingName: nil)
+        XCTAssertEqual(res.mimeType, mimetype, "should be the other mimetype")
+    }
+    
+    func test_ExpectedContentLength_positive() {
+        let contentLength = 100
+        let res1 = URLResponse(url: testURL, mimeType: "text/plain", expectedContentLength: contentLength, textEncodingName: nil)
+        XCTAssertEqual(res1.expectedContentLength, Int64(contentLength), "should be positive Int64 content length")
```
Comment (spevans) [inline TestFoundation/TestURLResponse.swift]: Has the test with `zeroContentLength` been removed? In general we don't remove tests unless they are invalid and not required anymore, so I would just add back in the zero content length test to this test function.

**Comment by matis-schotte** [comment]
Yes, I did change it to -1 but going to add the 0 test back in, thanks for the reply!

File: TestFoundation/TestURLResponse.swift (hunk 1 above)
Comment (matis-schotte) [reply TestFoundation/TestURLResponse.swift]: I took it back in :)
But as a separate one; as I've seen in the other PR you usually prefer less tests in favor of a bit longer ones? Hope its fine, but of course if required could merge them together.

**Comment by matis-schotte** [comment]
Could we try the ci-tests again?

**Comment by spevans** [comment]
@swift-ci test

**Comment by ikesyo** [comment]
@swift-ci test

File: TestFoundation/TestURLResponse.swift [hunk 2]
```diff
@@ -11,56 +11,63 @@ class TestURLResponse : XCTestCase {
...
-        let url = URL(string: "test")!
-        let res1 = URLResponse(url: url, mimeType: nil, expectedContentLength: 0, textEncodingName: encoding)
-        XCTAssertEqual(res1.textEncodingName, encoding, "should be the utf8 encoding")
-        let res2 = URLResponse(url: url, mimeType: nil, expectedContentLength: 0, textEncodingName: nil)
-        XCTAssertNil(res2.textEncodingName)
+        var encoding = "utf8"
+        var res = URLResponse(url: testURL, mimeType: nil, expectedContentLength: 0, textEncodingName: encoding)
+        XCTAssertEqual(res.textEncodingName, encoding, "should be the utf8 encoding")
+        
+        let res = URLResponse(url: testURL, mimeType: nil, expectedContentLength: 0, textEncodingName: nil)
```
Comment (ikesyo) [inline TestFoundation/TestURLResponse.swift]: This did not compile:

> /home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux/swift-corelibs-foundation/TestFoundation/TestURLResponse.swift:66:13: error: definition conflicts with previous value

Comment (matis-schotte) [reply TestFoundation/TestURLResponse.swift]: Very sorry to waste your time, this should not have happened 🙇‍♂️ I have setup Mojave now and can directly develop and test on the master branch.

It's fixed, please ci-re-test? 🙏

**Comment by ikesyo** [comment]
@swift-ci test

**Comment by matis-schotte** [comment]
Is it possible that sometimes the CI doesn't execute? Seems it's still waiting for the linux swift test even though the test was triggered a day ago.

**Comment by ikesyo** [comment]
@swift-ci test

**Pull request merged**